lib/worker/src/rpc.rs

```diff
@@ -207,21 +214,27 @@ impl<E: EVMClient + 'static> publications::Server for Publications<E> {
         if rel.is_empty() {
             return Promise::err(Error::failed("relation is required".into()));
         }
-        let limit = args.get_limit();
-        let offset = args.get_offset();
+        let limit = args.get_limit() as i32;
+        let offset = args.get_offset() as i32;
 
-        let e = self.evm_client.clone();
+        let pg_pool = self.pg_pool.clone();
+        let web3storage_client = self.web3storage_client.clone();
         Promise::from_future(async move {
-            let deals = e
-                .deals(format!("{}.{}", ns, rel).as_str(), offset, limit)
-                .await?;
+            let cids = db::pub_cids(&pg_pool, ns, rel, limit, offset).await?;
 
-            let mut deals_list = results.get().init_deals(deals.len() as u32);
-            for (i, d) in deals.iter().enumerate() {
+            let mut deals_list = results.get().init_deals(cids.len() as u32);
+            for (i, cid) in cids.iter().enumerate() {
                 let mut builder = deals_list.reborrow().get(i as u32);
-                builder.set_id(d.id);
-                builder.set_cid(d.cid.as_str().into());
-                builder.set_selector_path(d.selector_path.as_str().into());
+
+                let status = web3storage_client
```

Comment:
for each cid we do an HTTP call to web3 storage to fetch. this is not ideal (an HTTP call inside a loop). but i guess this is fine now as a temporary solution

Reply:
possible to await all requests at the same time? looks like they are awaited serially.

Reply:
maybe with `tokio::join!`: https://docs.rs/tokio/latest/tokio/macro.join.html#examples

Reply:
sounds good! will change it to do the calls asynchronously